test/mock-agent.js

```diff
@@ -2463,3 +2463,33 @@ test('MockAgent - using fetch yields a headers object in the reply callback', {
 
   t.end()
 })
+
+// https://github.com/nodejs/undici/issues/1579
+test('MockAgent - headers in mock dispatcher intercept should be case-insensitive', { skip: nodeMajor < 16 }, async (t) => {
+  const { fetch } = require('..')
+
+  const mockAgent = new MockAgent()
```

Comment:
```suggestion
  const mockAgent = new MockAgent()
  mockAgent.disableNetConnect()
```

If the request does not have a corresponding interceptor, the test wouldn't fail (it would instead make an actual http request to `example.com` and would probably succeed). Setting disableNetConnect will cause an error to be thrown if this request isn't mocked.

Reply:
Good catch. Thanks! [956023c](https://github.com/nodejs/undici/pull/1585/commits/956023c9bc0e3b47f5f8b1f4fa30ff6e1b115a06)